packages/date-picker/src/vaadin-date-picker-mixin.d.ts

```diff
@@ -237,6 +237,11 @@ export declare class DatePickerMixinClass {
    */
   max: string | undefined;
 
+  /**
+   * A function that is used to determine if a date should be disabled.
+   */
+  isDateDisabled: (date: Date) => boolean;
```

Comment:
Public API shouldn't deal with `Date` objects but the same ISO 8601 strings as `value`. This is to avoid timezone-related issues.

Reply:
I do understand the timezone issue, but presently the `dateAllowed` function that ends up calling the new `isDateDisabled` function is fed a `Date` object. So it seems like an issue further upstream from this change. Thoughts?

Reply:
The component uses `Date` a lot internally. There's a helper `_formatISO` that's used to convert a `Date` into an ISO string date for user-facing API. I'm not sure if `isDateDisabled` should use ISO dates or similar objects as `formatDate` and `parseDate` do.

Reply:
Ok, I think I'm getting you here. To sum up: The function the user provides should be working from either an ISO date or the `DatePickerDate` object like `formatDate` does. That keeps our user-land functions away from the problems of Javascript dates. So that means inside `dateAllowed` I'll need to parse the incoming `Date` object into ISO or `DatePickerDate` and pass that to the incoming function.

Reply:
One more followup as I implement this...

The `DatePickerDate` object doesn't give us quite as much functionality as the native `Date`, namely in regards to things like current day of the week (`date.getDay()`). So if a user wants to exclude weekends using `DatePickerDate` their function will have to convert it back into a fullblown `Date` just to get that functionality. I'm going to proceed and just alter my examples, but it does seem to be a more limiting type. We could go with an ISO string and then we just leave them to again convert it to a `Date` and then get the needed functionality.